### PR TITLE
Candidates Model / Router

### PR DESCRIPTION
**Pull request by BDesselle**
# Description

Please include a summary of the change and a link to which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change status
- [ ] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:
Everything in this checklist is required for each PR.  Please do not approve a PR that does not have all of these items.  

- [ ] I have pulled from master before submitting this PR
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer (the peer review to approve a PR counts.  The reviewer must download and test the code)
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] There are no merge conflicts

### CODE REVIEW

File: api/models/candidates-model.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+const db = require('../../data/dbConfig');
+
+async function getCandidates() {
```
Comment (ok1a) [inline api/models/candidates-model.js]: remove async

File: api/models/candidates-model.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  return db('candidates');
+}
+
+function getCandidatesById(id) {
+  return db('candidates')
+    .where({ id })
+    .first();
+}
+
+async function addCandidate(candidate) {
```
Comment (ok1a) [inline api/models/candidates-model.js]: either remove async or add a function to perform after insert.
if you were to chain a getCandidateById after the insert, then you would need it to be an async function.

File: api/routes/candidatesRouter.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+router.post('/', async (req, res) => {
+  Candidates.addCandidate(req.body)
+    .then(candidate => {
+      res.status(200).json({
+        message: `Candidate was successfully added to database.`,
+        id: candidate[0],
+      });
+    })
+    .catch(error => {
+      if (error.code === 'SQLITE_CONSTRAINT') {
```
Comment (meginks) [inline api/routes/candidatesRouter.js]:
```suggestion
```

File: api/routes/candidatesRouter.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        return res.status(500).json({
```
Comment (meginks) [inline api/routes/candidatesRouter.js]:
```suggestion
```

File: api/routes/candidatesRouter.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+          message: 'Candidate is already in the database. Proceed like normal.',
```
Comment (meginks) [inline api/routes/candidatesRouter.js]:
```suggestion
```

File: api/models/candidates-model.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  return db('candidates').insert(candidate, 'id');
+}
+
+function updateCandidate(id, candidate) {
+  return db('candidates')
+    .where('id', Number(id))
```
Comment (ok1a) [inline api/models/candidates-model.js]: Should not need Number()

File: api/routes/candidatesRouter.js [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+        });
```
Comment (meginks) [inline api/routes/candidatesRouter.js]:
```suggestion
```

File: api/routes/candidatesRouter.js [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+      }
```
Comment (meginks) [inline api/routes/candidatesRouter.js]:
```suggestion

```

File: api/routes/candidatesRouter.js (hunk 3 above)
Comment (meginks) [inline api/routes/candidatesRouter.js]:
```suggestion
```

File: api/routes/candidatesRouter.js [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+      res.status(404).json({ message: 'No candidate found with that ID' });
+    }
+  } catch (err) {
+    res.status(500).json(err);
+  }
+});
+
+router.post('/', async (req, res) => {
+  Candidates.addCandidate(req.body).then(error => {
+    if (error.code === 'SQLITE_CONSTRAINT') {
```
Comment (meginks) [inline api/routes/candidatesRouter.js]: remove lines 36-40

File: api/routes/candidatesRouter.js [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+    } else {
+      res.status(404).json({ message: 'No candidate found with that ID' });
+    }
+  } catch (err) {
+    res.status(500).json(err);
+  }
+});
+
+router.post('/', async (req, res) => {
+  Candidates.addCandidate(req.body).then(error => {
```
Comment (ok1a) [inline api/routes/candidatesRouter.js]: I believe this should be .then(response object=>{
res..etc(response object)}).catch(err=>{ etc ...res..etc(err)})

File: api/models/candidates-model.js [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+    .first();
+}
+
+function addCandidate(candidate) {
+  return db('candidates').insert(candidate, 'id');
+}
+
+function updateCandidate(id, candidate) {
+  return db('candidates')
+    .where('id')
```
Comment (ok1a) [inline api/models/candidates-model.js]: .where('id').first().update(etc)

File: api/models/candidates-model.js [hunk 12]
```diff
@@ -16,8 +16,8 @@ function addCandidate(candidate) {
 
 function updateCandidate(id, candidate) {
   return db('candidates')
-    .where('id')
-    .update(candidate);
+    .update(candidate)
+    .where('id');
```
Comment (ok1a) [inline api/models/candidates-model.js]: a .first()' should fix this

**Comment by meginks** [comment]
Working!

**Pull request merged**